frontend/lib/resolution/prims.cpp

```diff
@@ -1332,14 +1331,31 @@ CallResolutionResult resolvePrimCall(Context* context,
                                IntParam::get(context, s));
           break;
         } else if (actualType.type()->isStringType() ||
-                   actualType.type()->isBytesType()) {
+                   actualType.type()->isBytesType() ||
+                   actualType.type()->isCStringType()) {
           // for non-param string/bytes, the return type is just a default int
           type = QualifiedType(QualifiedType::CONST_VAR,
                                IntType::get(context, 0));
           break;
         }
       }
     }
+    case PRIM_STRING_CONCAT:
+    {
+      if (ci.numActuals() == 2) {
+        auto lhs = ci.actual(0).type();
+        auto rhs = ci.actual(1).type();
+        CHPL_ASSERT(lhs.type() == rhs.type());
+        CHPL_ASSERT(lhs.isParam() && rhs.isParam());
+        CHPL_ASSERT(lhs.type()->isStringType() || lhs.type()->isBytesType());
```

Comment:
I strongly suggest not asserting here. If dyno is being used in a tool (LSP, linter, etc), merely encountering a bad concat call with crash that tool if assertions are enabled. I think we should avoid crashing as much as we can going forward. Consider leaving type unset if the conditions you assert aren't met.

Reply:
Good point, I'll get rid of them. I think I'll also open an issue somewhere about how we should enforce types when it comes to primitives.